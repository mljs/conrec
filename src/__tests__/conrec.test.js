import fs from 'fs';

import converter from 'jcampconverter';

import { Conrec } from '..';

const data = fs.readFileSync(`${__dirname}/data/zhmbc_0.jdx`, 'utf8');
const parsed = converter.convert(data, { noContour: true });

describe('conrec basic test', () => {
  it('no result because level too far', () => {
    const conrec = new Conrec(parsed.minMax.z);
    const basic = conrec.drawContour({
      contourDrawer: 'basic',
      levels: [-1000000000, 1000000000],
      timeout: 10000,
    });

    expect(basic).toStrictEqual([
      { lines: [], zValue: -1000000000 },
      { lines: [], zValue: 1000000000 },
    ]);
  });

  it('2 specified levels', () => {
    const conrec = new Conrec(parsed.minMax.z);
    const basic = conrec.drawContour({
      contourDrawer: 'basic',
      levels: [-100000, 100000],
      timeout: 10000,
    });

    expect(basic).toHaveLength(2);
    expect(basic[0].lines).toHaveLength(36864);
    expect(basic[1].lines).toHaveLength(119720);
  });

  it('auto select levels', () => {
    const conrec = new Conrec(parsed.minMax.z);
    const basic = conrec.drawContour({
      contourDrawer: 'basic',
      nbLevels: 10,
      timeout: 10000,
    });

    expect(basic).toHaveLength(10);
    expect(basic[0].lines).toHaveLength(0);
    expect(basic[1].lines).toHaveLength(4984);
    expect(basic[8].lines).toHaveLength(32);
    expect(basic[8].lines).toMatchSnapshot();
    expect(basic[9].lines).toHaveLength(0);
  });

  it('auto select levels with swapAxes', () => {
    const conrec = new Conrec(parsed.minMax.z, { swapAxes: true });
    const basic = conrec.drawContour({
      contourDrawer: 'basic',
      nbLevels: 10,
      timeout: 10000,
    });

    expect(basic).toHaveLength(10);
    expect(basic[0].lines).toHaveLength(0);
    expect(basic[1].lines).toHaveLength(4984);
    expect(basic[8].lines).toHaveLength(32);
    expect(basic[8].lines).toMatchSnapshot();
    expect(basic[9].lines).toHaveLength(0);
  });
});
