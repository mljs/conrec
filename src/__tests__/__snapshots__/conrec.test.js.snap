// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conrec basic test auto select levels 1`] = `
Array [
  839.6939653486645,
  1880.3060346513357,
  839.4964607464607,
  1880,
  840,
  1880.697601854328,
  839.6939653486645,
  1880.3060346513357,
  840.5295033157395,
  1880,
  840.3157788539144,
  1880.3157788539145,
  840.3157788539144,
  1880.3157788539145,
  840,
  1880.697601854328,
  839.6766862927384,
  1879.6766862927382,
  840,
  1879.178477690289,
  839.4964607464607,
  1880,
  839.6766862927384,
  1879.6766862927382,
  840.3376920421198,
  1879.66230795788,
  840.5295033157395,
  1880,
  840,
  1879.178477690289,
  840.3376920421198,
  1879.66230795788,
]
`;
